src/wallet/spend.cpp

```diff
@@ -1092,6 +1092,13 @@ util::Result<CreatedTransactionResult> CreateTransaction(
         TRACE1(coin_selection, attempting_aps_create_tx, wallet.GetName().c_str());
         CCoinControl tmp_cc = coin_control;
         tmp_cc.m_avoid_partial_spends = true;
+
+        // Re-use the change destination from the first creation attempt to avoid skipping BIP44 indexes
+        const int change_pos = txr_ungrouped.change_pos;
```

Comment:
nit: this shadows the other `change_pos` variable. Can overwrite it instead of create a new one.

Reply:
I think this is actually more than a nit, it can have consequences on the change position of the change output for the aps solution.

If the user specified a change position but we didn't use it, now `change_pos` would be `-1` , so the aps run would put its change in a random position, if it needs one. This could put the change in an unexpected position. So we shouldn't  overwrite nor shadow `change_pos`.

Good catch!

Reply:
ha, spider sense. Nice. Great catch for you!.

Orthogonal topic; moving forward, one of the follow-ups that have post-#25806 is the removal of this secondary "grouped" tx creation. We will be able to run Coin Selection directly over the aps vs non-aps grouped outputs.

Reply:
Yeah great catch thanks, sorry I didn't notice that already. I added commit 80ad44d10ca0831869a5e50c36e07e7ff10902ec which includes a test for the edge case:

- user requests tx with specific `change_position`
- wallet creates default tx that does not require change (`change_pos` is overwritten with `-1`)
- wallet creates APS tx as second attempt, that does require change (change is placed in random position)
- APS tx is selected and returned, change position is wrong

Looking for feedback on the new test before I squash...